### PR TITLE
feat: Move Create report to platform [5]

### PR DESCRIPTION
**Pull request by bistaastha**
PR content:
- Moves Create Report to platform
- Removes old create report dialog - this is not in use anymore and was causing issues while removing the createDraft method from reports service

### CODE REVIEW

File: src/app/core/services/platform/v1/spender/reports.service.spec.ts [hunk 1]
```diff
@@ -232,6 +233,7 @@ describe('SpenderReportsService', () => {
     spenderReportsService.createDraft(reportParam).subscribe((res) => {
       expect(res).toEqual(allReportsPaginated1.data[0]);
       expect(spenderPlatformV1ApiService.post).toHaveBeenCalledOnceWith('/reports', reportParam);
+      spyOn(spenderReportsService, 'clearTransactionCache').and.returnValue(of(null));
```
Comment (bistaastha) [inline src/app/core/services/platform/v1/spender/reports.service.spec.ts:236]: Fixed in follow up

File: src/app/core/services/platform/v1/spender/reports.service.ts [hunk 2]
```diff
@@ -73,6 +73,15 @@ export class SpenderReportsService {
     );
   }
 
+  @CacheBuster({
+    cacheBusterNotifier: reportsCacheBuster$,
+  })
+  createDraft(data: CreateDraftParams): Observable<Report> {
+    return this.spenderPlatformV1ApiService
+      .post<PlatformApiPayload<Report>>('/reports', data)
+      .pipe(switchMap((res) => this.clearTransactionCache().pipe(map(() => res.data))));
```
Comment (Chethan-Fyle) [inline src/app/core/services/platform/v1/spender/reports.service.ts:82]: Why nested pipe?

Comment (Chethan-Fyle) [reply src/app/core/services/platform/v1/spender/reports.service.ts:82]: Use `tap()` if required.

Comment (Chethan-Fyle) [reply src/app/core/services/platform/v1/spender/reports.service.ts:82]: The method name is `createDraft()` but I don't see anything in the definition that relates to draft. I see only the creation of a regular report.

File: src/app/core/services/report.service.ts [hunk 3]
```diff
@@ -130,18 +132,9 @@ export class ReportService {
...
-  }
-
-  @CacheBuster({
-    cacheBusterNotifier: reportsCacheBuster$,
-  })
-  create(report: ReportPurpose, expenseIds: string[]): Observable<ReportV1> {
-    return this.createDraft(report).pipe(
-      switchMap((newReport: ReportV1) => {
+  create(report: ReportPurpose, expenseIds: string[]): Observable<Report> {
+    return this.spenderReportsService.createDraft({ data: report }).pipe(
```
Comment (Chethan-Fyle) [inline src/app/core/services/report.service.ts:136]: There we have create() method and inside we are calling `createDraft()`. This can be refactored IMO.

**Review by Chethan-Fyle — CHANGES_REQUESTED** [review]
Few comments.

**Review by mvaishnavi — APPROVED** [review]
another review from me not needed.

File: src/app/core/services/platform/v1/spender/reports.service.ts (hunk 2 above)
Comment (bistaastha) [reply src/app/core/services/platform/v1/spender/reports.service.ts:82]: This method creates draft reports, for a submitted report this API is used in combination with `submit` API from public

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**